projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/data-job/pages/executions/time-period-filter/time-period-filter.component.ts

```diff
@@ -0,0 +1,289 @@
+import moment from 'moment';
+
+import { CollectionsUtil } from '@vdk/shared';
+
+type CustomFormGroup = FormGroup & { controls: { [key: string]: FormControl } };
+
+@Component({
+    selector: 'lib-time-period-filter',
+    templateUrl: './time-period-filter.component.html',
+    styleUrls: ['./time-period-filter.component.scss']
+})
+export class TimePeriodFilterComponent implements OnInit, OnDestroy {
+    @ViewChild('fromPicker') fromPicker: DatePickerDirective;
+    @ViewChild('toPicker') toPicker: DatePickerDirective;
+
+    @Input() set loading(value: boolean) {
+        this._loading = value;
+    }
+
+    get loading() {
+        return this._loading;
+    }
+
+    @Output() filterChanged: EventEmitter<{ fromTime: Date; toTime: Date }> = new EventEmitter<{ fromTime: Date; toTime: Date }>();
+
+    pickerConfig: IDatePickerDirectiveConfig = {
+        // TODO [format: 'MMM DD, YYYY, hh:mm:ss A',] dayjs
+        format: 'MMM DD, yyyy, hh:mm:ss A',
+        showGoToCurrent: true,
+        showTwentyFourHours: false,
+        showSeconds: true,
+        weekDayFormat: 'dd',
+        numOfMonthRows: 6,
+        monthBtnFormat: 'MMMM'
+    };
+    fromPickerConfig: IDatePickerDirectiveConfig = {
+        ...this.pickerConfig
+    };
+    toPickerConfig: IDatePickerDirectiveConfig = {
+        ...this.pickerConfig
+    };
+
+    fromTime: Date;
+    toTime: Date;
+
+    fromTimeMin: moment.Moment;
+    fromTimeMax: moment.Moment;
+    toTimeMin: moment.Moment;
+    toTimeMax: moment.Moment;
+
+    tmForm: CustomFormGroup;
+
+    @Input()
+    set minTime(date: Date) {
+        if (CollectionsUtil.isNil(date)) {
+            return;
+        }
+
+        if (!this._initiallySetMin) {
+            this._initiallySetMin = true;
+
+            this.fromTime = date;
+            this._minTimeOrig = date;
+
+            const minDateTime = this._normalizeDateTime(date, 'min', 'utc');
+
+            this.fromTimeMin = minDateTime;
+            this.fromPickerConfig = {
+                ...this.fromPickerConfig,
+                min: this.fromTimeMin
+            };
+
+            this.toTimeMin = minDateTime;
+            this.toPickerConfig = {
+                ...this.toPickerConfig,
+                min: this.toTimeMin
+            };
+
+            if (!this._isFromPickerOpened) {
+                this.tmForm
+                    .get('fromDate')
+                    .patchValue(
+                        this._normalizeDateTime(date, 'from', 'utc')
+                            .format(this.fromPickerConfig.format)
+                    );
+            }
+        }
+
+        this._refreshMaxTime();
+    }
+
+    private _loading = false;
+
+    private _minTimeOrig: Date;
+
+    private _isFromPickerOpened = false;
+    private _isToPickerOpened = false;
+    private _initiallySetMin = false;
+    private _initiallySetMax = false;
+    private _refreshIntervalRef: number;
+
+    /**
+     * ** Constructor.
```

Comment:
Nit: This is pretty useless 😄

Reply:
nit: yeah...may be auto-generated; it's very nit, we can refactor while code is evolving?